src/main/java/org/jenkinsci/plugins/cloudstats/ProvisioningActivity.java

```diff
@@ -323,17 +324,34 @@ public long getStartedTimestamp() {
      *
      * @throws IllegalArgumentException In case phases are entered repeatedly.
      */
+    @SuppressFBWarnings(value="RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE")
     public void enter(@Nonnull Phase phase) {
         synchronized (progress) {
             if (progress.get(phase) != null) throw new IllegalStateException(
                     "The phase " + phase + " has already started"
             );
 
-            if (getCurrentPhase().compareTo(phase) >= 0) throw new IllegalStateException(
+            final Phase currentPhase = getCurrentPhase();
+            if (currentPhase.compareTo(phase) >= 0) throw new IllegalStateException(
                     "The phase " + getCurrentPhase() + " has already started"
             );
 
             progress.put(phase, new PhaseExecution(phase));
+
+            if (phase.compareTo(Phase.COMPLETED) == 0 && !currentPhase.equals(Phase.OPERATING)) {
+                PhaseExecution pe = progress.get(Phase.PROVISIONING);
+                if (pe == null || (pe != null && pe.getStatus().equals(Status.OK))) {
```

Comment:
In `x == null || ...`, `x` is guaranteed not to be `null` on the right side due to short-cirquiting.